test/integration/http2_integration_test.cc

```diff
@@ -489,6 +489,36 @@ TEST_P(Http2RingHashIntegrationTest, CookieRoutingNoCookieWithTtlSet) {
   EXPECT_EQ(set_cookies.size(), 1);
 }
 
+TEST_P(Http2RingHashIntegrationTest, CookieRoutingNoCookieWithZeroTtlSet) {
+  config_helper_.addConfigModifier(
+      [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm)
+          -> void {
+        auto* hash_policy = hcm.mutable_route_config()
+                                ->mutable_virtual_hosts(0)
+                                ->mutable_routes(0)
+                                ->mutable_route()
+                                ->add_hash_policy();
+        auto* cookie = hash_policy->mutable_cookie();
+        cookie->set_name("foo");
+        cookie->mutable_ttl();
+      });
+
+  std::set<std::string> set_cookies;
+  sendMultipleRequests(
+      1024,
+      Http::TestHeaderMapImpl{{":method", "POST"},
+                              {":path", "/test/long/url"},
+                              {":scheme", "http"},
+                              {":authority", "host"}},
+      [&](IntegrationStreamDecoder& response) {
+        EXPECT_STREQ("200", response.headers().Status()->value().c_str());
+        std::string value = response.headers().get(Http::Headers::get().SetCookie)->value().c_str();
+        set_cookies.insert(value);
+        EXPECT_THAT(value, MatchesRegex("^foo=.*$"));
+      });
+  EXPECT_EQ(set_cookies.size(), 1);
```

Comment:
Do you want this to be a vector (or multiset) so you can validate the cardinality matches the number of requests?

Reply:
I could change it to a `std::map<string, size_t>` and increment the count for the key, but the `sendMultipleRequests` helper already EXPECTs that each request is completed and calls this callback for each one.

Reply:
True, I think this comment can be ignored. Ready to merge.